src/panels/lovelace/cards/hui-statistics-graph-card.ts

```diff
@@ -73,6 +79,20 @@ export class HuiStatisticsGraphCard extends LitElement implements LovelaceCard {
 
   private _statTypes?: Array<StatisticType>;
 
+  public hassSubscribe() {
+    if (!this._config?.energy_date_selection) {
+      return [];
+    }
+
+    return [
+      getEnergyDataCollection(this.hass!, {
+        key: this._config?.collection_key,
+      }).subscribe((data) => {
+        this._setFetchStatisticsTimer(data);
```

Comment:
You should not pass the data to this function, as the function is also called in other places, save the data you need (the dates) in the class and then call the `_setFetchStatisticsTimer` function.